src/main/java/com/github/steveice10/mc/protocol/data/message/MessageSerializer.java

```diff
@@ -64,7 +62,11 @@ public static JsonElement toJson(Message message) {
         if(message instanceof TextMessage && message.getStyle().equals(MessageStyle.DEFAULT) && message.getExtra().isEmpty()) {
             return new JsonPrimitive(((TextMessage) message).getText());
         }
+        
+        return toJsonObject(message);
+    }
 
+    public static JsonObject toJsonObject(Message message) {
```

Comment:
Is there a reason this was split off?

Reply:
If you need to get a JsonObject all the time instead of sometimes getting a JsonPrimitive this allows for that, as you cant construct it yourself because the methods used are private.

Reply:
Yes, but I don't understand what situation you'd specifically need a JsonObject in over the simpler representation.

Reply:
It was for getting around the parent style issue, see here: https://github.com/GeyserMC/Geyser/blob/feature/1.16/connector/src/main/java/org/geysermc/connector/utils/MessageUtils.java#L208-L238

Reply:
Instead of messing with the raw JSON, it would probably be a better solution if I added a method for copying an existing message into a builder, so you could just do something like:
```
message.toBuilder()
    .style(message.getStyle().toBuilder()
                 .color(parent.getStyle().getColor())
                 .formats(parent.getStyle().getFormats()))
    .build()
```

I could get that put together after the PR is merged if you want.

Reply:
That would be great, but would it be possible to get that PRd onto this PR? So I can just Geyser to work with that before this is merged to prevent any delays releasing.

Reply:
I've committed it to the main repository, so you can go ahead and pull it into your branch.

Reply:
This should be fixed now